demo/src/app/examples/advanced/datatable-integration/datatable.type.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, OnInit, Input, ViewChild, TemplateRef } from '@angular/core';
+import { FormlyFieldConfig, FieldArrayType, FormlyFormBuilder } from '@ngx-formly/core';
+import { TableColumn } from '@swimlane/ngx-datatable/release/types';
+
+
+@Component({
+  selector: 'app-table-type',
+  template: `
+<ngx-datatable
+  #table  class="bootstrap" 
+  [rows]="model"
+  [columns]="columns"
+  [columnMode]="columnMode"
+  [rowHeight]="rowHeight"   
+  [headerHeight]="headerHeight"      
+  [footerHeight]="footerHeight"
+  [limit]="limit"  
+  [scrollbarH]="scrollbarH"      
+  [reorderable]="reorderable"    
+  [externalSorting]="true"
+  [selected]="selected"
+  [selectionType]="'single'"
+  (sort)="onSort($event)"
+  (select)='onSelect($event)'>     
+  
+  <ng-template #genericcolumn ngx-datatable-cell-template let-rowIndex="rowIndex" let-value="value" let-row="row" let-column="column">
+    <formly-field             
+      [model]="getModel(model,column,rowIndex)"
+      [field]="getFields(field,column, rowIndex)"
+      [options]="options"
+      [form]="formControl">
+    </formly-field>
+  </ng-template>  
+  
+</ngx-datatable>
+
+`
+})
+
+export class DatatableTypeComponent extends FieldArrayType {
+  constructor(builder: FormlyFormBuilder) {
+    super(builder);
+  }
+
+  @ViewChild('genericcolumn') public genericcolumn: TemplateRef<any>;
+  
+  @Input() columnMode: any = "force";    
```

Comment:
I would prefer to move these options into `defaultOptions` declaration, see the datepicker example [datepicker.module.ts#L26-L32](https://github.com/formly-js/ngx-formly/blob/master/src/material/datepicker/src/datepicker.module.ts#L26-L32)

with that way those option can be passed through `fields`

Reply:
ok thanks